From 852615b0df2252ea67f4f82e9ace2fc2794467b3 Mon Sep 17 00:00:00 2001
From: Marcus Lundblad <ml@dfupdate.se>
Date: Thu, 7 Mar 2024 23:34:01 +0100
Subject: [PATCH] tests: Explicitly enable assertion macros

To allow building with G_DISABLE_ASSERT we need to explicitly
undefine it when building the tests, otherwise g_test_init() turns
into an abort since GLib 2.58.

See: https://gitlab.freedesktop.org/dbus/dbus-glib/-/merge_requests/4
---
 tests/coordinate.c          | 2 ++
 tests/data-source-request.c | 2 ++
 tests/file-cache.c          | 3 ++-
 tests/license.c             | 2 ++
 tests/location.c            | 2 ++
 tests/map.c                 | 2 ++
 tests/marker-layer.c        | 2 ++
 tests/marker.c              | 2 ++
 tests/memory-cache.c        | 2 ++
 tests/vector-expression.c   | 3 ++-
 tests/vector-index.c        | 2 ++
 tests/vector-reader.c       | 2 ++
 tests/vector-renderer.c     | 2 ++
 tests/vector-sprite-sheet.c | 3 ++-
 tests/vector-style.c        | 2 ++
 tests/vector-value.c        | 2 ++
 tests/viewport.c            | 2 ++
 17 files changed, 34 insertions(+), 3 deletions(-)

diff --git a/tests/coordinate.c b/tests/coordinate.c
index aeff8978..3c847f58 100644
--- a/tests/coordinate.c
+++ b/tests/coordinate.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/data-source-request.c b/tests/data-source-request.c
index 785ec857..60ce7bfe 100644
--- a/tests/data-source-request.c
+++ b/tests/data-source-request.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/file-cache.c b/tests/file-cache.c
index 721133a0..9b981dcd 100644
--- a/tests/file-cache.c
+++ b/tests/file-cache.c
@@ -1,5 +1,6 @@
-#include <shumate/shumate.h>
+#undef G_DISABLE_ASSERT
 
+#include <shumate/shumate.h>
 
 #define TEST_ETAG "0123456789ABCDEFG"
 #define TEST_DATA "The quick brown fox \0 jumps over the lazy dog"
diff --git a/tests/license.c b/tests/license.c
index 4b7caded..1bd677a3 100644
--- a/tests/license.c
+++ b/tests/license.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/location.c b/tests/location.c
index 6c288724..0dc67993 100644
--- a/tests/location.c
+++ b/tests/location.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <shumate/shumate.h>
 
 void
diff --git a/tests/map.c b/tests/map.c
index 32d962ca..925c1f4e 100644
--- a/tests/map.c
+++ b/tests/map.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/marker-layer.c b/tests/marker-layer.c
index f075c134..793e2045 100644
--- a/tests/marker-layer.c
+++ b/tests/marker-layer.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/marker.c b/tests/marker.c
index 94e68615..0fc3b3b4 100644
--- a/tests/marker.c
+++ b/tests/marker.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/memory-cache.c b/tests/memory-cache.c
index 7e96933e..550db59f 100644
--- a/tests/memory-cache.c
+++ b/tests/memory-cache.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <shumate/shumate.h>
 #include "shumate/shumate-memory-cache-private.h"
 
diff --git a/tests/vector-expression.c b/tests/vector-expression.c
index f4437dad..9e258fbb 100644
--- a/tests/vector-expression.c
+++ b/tests/vector-expression.c
@@ -1,9 +1,10 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 #include "shumate/vector/shumate-vector-expression-interpolate-private.h"
 #include "shumate/vector/shumate-vector-expression-filter-private.h"
 
-
 static void
 test_vector_expression_parse (void)
 {
diff --git a/tests/vector-index.c b/tests/vector-index.c
index f6c1bc47..a9ab3035 100644
--- a/tests/vector-index.c
+++ b/tests/vector-index.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <shumate/shumate.h>
 #include "shumate/vector/shumate-vector-index-private.h"
 #include "shumate/vector/shumate-vector-render-scope-private.h"
diff --git a/tests/vector-reader.c b/tests/vector-reader.c
index 03dedd7d..81c221b3 100644
--- a/tests/vector-reader.c
+++ b/tests/vector-reader.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <shumate/shumate.h>
 #include "shumate/vector/vector_tile.pb-c.h"
 
diff --git a/tests/vector-renderer.c b/tests/vector-renderer.c
index bb6a3a68..c8bea344 100644
--- a/tests/vector-renderer.c
+++ b/tests/vector-renderer.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <glib-object.h>
 #include <shumate/shumate.h>
 #include "shumate/shumate-tile-private.h"
diff --git a/tests/vector-sprite-sheet.c b/tests/vector-sprite-sheet.c
index 58f11f4c..ad93e339 100644
--- a/tests/vector-sprite-sheet.c
+++ b/tests/vector-sprite-sheet.c
@@ -1,7 +1,8 @@
+#undef G_DISABLE_ASSERT
+
 #include <shumate/shumate.h>
 #include "shumate/shumate-vector-sprite-sheet.h"
 
-
 static void
 test_vector_sprite_sheet (void)
 {
diff --git a/tests/vector-style.c b/tests/vector-style.c
index c99ea9c5..af00df48 100644
--- a/tests/vector-style.c
+++ b/tests/vector-style.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
diff --git a/tests/vector-value.c b/tests/vector-value.c
index 1a5880c4..fe132603 100644
--- a/tests/vector-value.c
+++ b/tests/vector-value.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 #include "shumate/vector/shumate-vector-value-private.h"
diff --git a/tests/viewport.c b/tests/viewport.c
index b38dbb7e..9de66e7f 100644
--- a/tests/viewport.c
+++ b/tests/viewport.c
@@ -1,3 +1,5 @@
+#undef G_DISABLE_ASSERT
+
 #include <gtk/gtk.h>
 #include <shumate/shumate.h>
 
-- 
GitLab

